src/Features/LanguageServer/Protocol/Handler/ResolveCache.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler;
+
+/// <summary>
+/// A common LSP pattern is an initial request to the server that returns some set of partially filled out items.
+/// Then the client issues a xyz/resolve request to fully resolve a specific item when actually needed.
+/// 
+/// On the server side we often need to cache non-serializable data that can't be included in the typical
+/// 'data' field on the actual item.  This type is a general cache that helps keep track of data between requests.
+/// This is a general cache that helps perform that function.
+/// </summary>
```

Comment:
can you doc that the expected form is for the intiial request to populate this, and then later resolve requests will just read. (basically doc who should write and who should read).